contents/ruby/Rails.md

```diff
@@ -0,0 +1,154 @@
+
+## Why Rails
+
+Many popular and very technically mature websites started off using, and continue to use Rails. These include: GitHub, Airbnb, Twitter, Hulu, Shopify, and Twitch, among others. Some of the key features that makes Rails the framework of choice for these users are:
+
+### Fast iteration speed for product developement 
+
+As mentioned before, Rails is an opinionated framework that does things the "Rails" way. This includes automating trivial tasks by following certain conventions, which can make the development process faster. For example, in a Rails project that is connected to a database, each class is mapped to a table, as described in the Active Record Pattern, and the table name is just a pluralised version of the class's name - the `User` class becomes the `Users` table, the `Person` class becomes the `People` table, etc. This means that you don't need to spend any time deliberating on the name of the tables, nor do you have to juggle multiple concepts while mapping out the database and application, as Rails will automatically connect the two and spare you from deliberating about the exact name of the table. A similar approach is in seen in the usage of resources, since it also automates much of the process.
+
+The automation of these mundane tasks could make development a more enjoyable (more on that in a bit), and faster process. All of these optimizations can make developing an application between 30-40% faster on Rails, while also making it easier to learn, as you do not need to understand HTTP verbs to make a basic Rails application (while mastery of the topic will certainly help). Furthermore, programmers experienced in Rails usually find it easier to start on existing Rails projects, since the conventions followed are usually the same, giving you the same directory structure and route naming conventions everywhere. 
+  
+### Optimized for Programmer Happiness
+
+One of the primary tenets of Ruby was to make it the "Least Surprising Language", where the language was designed to perform exactly as the developer thought would be the most "expected" function. Rails built on that, and was designed to be optimized for Programmer Happiness. This meant re-imagining many of coding's most standard conventions, as well as adding significant bloat, all for the purpose of making the code look "pretty", and inherently "nice" to write (as subjective as that sounds). 
```

Comment:
I think

`One of the primary tenets of Ruby was to make it the "Least Surprising Language", where the language was designed to feel intuitive and perform exactly as a developer would expect. In Rails, even more syntactic sugar is added on top of its Ruby foundations, so that working with a Rails application would be a joyful experience.`

is a better set up for the examples that you are including later on. One of the gripes (among a few other issues) I have about this paragraph is that `adding significant bloat` feels out of place with what you're trying to convey.